client/utils/components.js

```diff
@@ -0,0 +1,53 @@
+import {settings} from './env'
+
+/**
+ * Calculate number of elements in array less than or equal to target values
+ *
+ * @param  {Number[]} arr     A sorted array of numbers
+ * @param  {[Number]} target  The number to be less than or equal to
+ * @return {int}              The number of elements in the array less than or equal to the target value
+ */
+export function calcNumLessThan (arr, target) {
```

Comment:
well, binary search is O(log(n)) vs O(n) for the filter method, so binary search is always going to be faster in the worst case for array length > 3.  This method is used primarily in the possibilities analysis componenet.  That component has a slider which can call this function a lot, so performance is important here.